### PR TITLE
initial pass to replace nLua with moonscript

### PR DESCRIPTION
**Pull request by baughj**
## Description
Remove nLua and replace with MoonScript, which is entirely written in C# with no weird DLL dependencies. This replacement is fairly straightforward with not many changes needed.

## Related PRs
n/a

## Checklist
- [x] Tested and working locally
- [ ] Reviewed
- [ ] Tested and working on dev server
- [ ] Deployed to staging

## How to QA
Run the server and try to execute lua scripts.

## Impacted Areas in Hybrasyl
All scripting.

### CODE REVIEW

File: hybrasyl/Scripting/Script.cs [hunk 1]
```diff
@@ -103,14 +105,16 @@ public dynamic GetObjectWrapper(WorldObject obj)
...
-            State.DoString(@"Enums = CLRPackage('Hybrasyl', 'Hybrasyl.Enums')");
-            State["world"] = Processor.World;
-            State["logger"] = new ScriptLogger(Name);
+
+            // Create index of functions
+         //   State.LoadCLRPackage();
+           // State.DoString(@"Scripting = CLRPackage('Hybrasyl', 'Hybrasyl.Scripting')");
+            //State.DoString(@"Enums = CLRPackage('Hybrasyl', 'Hybrasyl.Enums')");
+            //State["world"] = Processor.World;
+            //State["logger"] = new ScriptLogger(Name);
```
Comment (norrismiv) [inline hybrasyl/Scripting/Script.cs]: is this dead code? if so let's remove.

File: hybrasyl/Scripting/Script.cs [hunk 2]
```diff
@@ -103,14 +105,16 @@ public dynamic GetObjectWrapper(WorldObject obj)
...
             // Prohibit future imports from other .NET assemblies
-            State.DoString(@"import = function() end");
-            State.DoString(@"CLRPackage = function() end");
+            //State.DoString(@"import = function() end");
+            //State.DoString(@"CLRPackage = function() end");
```
Comment (norrismiv) [inline hybrasyl/Scripting/Script.cs]: more dead?

File: hybrasyl/Scripting/ScriptProcessor.cs [hunk 3]
```diff
@@ -22,7 +22,6 @@
  
 using System.Collections.Generic;
 using log4net;
-using NLua;
```
Comment (norrismiv) [inline hybrasyl/Scripting/ScriptProcessor.cs]: we may need this in the future

**Comment by norrismiv** [comment]
+1

File: hybrasyl/Scripting/Script.cs (hunk 2 above)
Comment (baughj) [reply hybrasyl/Scripting/Script.cs]: I was leaving this with the expectation of revisiting but yes. dead

**Pull request merged**